### PR TITLE
Speed up review app installation

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

Reduce the time to create the preview PR from ~11m to ~3m.

> [!NOTE]
> I plan to drop e4adf4f5f2d348b447d44bad64644fb5443d1ae3 since it doesn't seem to have any effect: compare runs [before](https://github.com/nextstrain/auspice/actions/runs/11410864785/job/31754024500) and [after](https://github.com/nextstrain/auspice/actions/runs/11410911840/job/31754172608)
> EDIT: [keeping it](https://github.com/nextstrain/auspice/pull/1870#issuecomment-2426977204) as 2aba0cf9c320ca154484069435bb5ff7074ed2ee

## Related issue(s)

- #1709

## Checklist

- [x] Checks pass
- [x] ~If making user-facing changes, add a message in [CHANGELOG.md](https://github.com/nextstrain/auspice/blob/HEAD/CHANGELOG.md) summarizing the changes in this PR~
- [x] (to be done by a Nextstrain team member) [Create preview PRs on downstream repositories][1].

[1]: https://github.com/nextstrain/auspice/blob/-/DEV_DOCS.md#test-on-downstream-repositories

### CODE REVIEW

File: .github/workflows/preview_on_downstream_repo.yaml
Comment (victorlin) [inline .github/workflows/preview_on_downstream_repo.yaml:1]: Aside: Adding the label before opening the PR causes weird concurrency issues. See how these two were triggered at the same time:

image

Links:

- [canceled](https://github.com/nextstrain/auspice/actions/runs/11395018611)
- [running](https://github.com/nextstrain/auspice/actions/runs/11395018603)

Comment (victorlin) [reply .github/workflows/preview_on_downstream_repo.yaml:1]: I thought they were different but they're actually the same. So this isn't really an issue and the concurrency rule is cancelling one as expected.

**Review by jameshadfield — APPROVED** [review]
Why drop https://github.com/nextstrain/auspice/commit/e4adf4f5f2d348b447d44bad64644fb5443d1ae3? Its commit message provides a compelling reason to keep it...

**Comment by victorlin** [comment]
@jameshadfield I forgot to explain in detail – I added that commit thinking it would ignore installation of Auspice `devDependencies`, but it only applies to `devDependencies` in the current project, i.e. [nextstrain.org/auspice-client/package.json](https://github.com/nextstrain/nextstrain.org/blob/7617c384f3aa8fb3083c46a5c29f9f90db8c2558/auspice-client/package.json) which has no `devDependencies` at the moment. You have a good point though – if that every changes in the future, we would still want to ignore `devDependencies` during installation for the review app. I'll keep the commit with a revised message.

Side note: the reason that Auspice `devDependencies` were installed and causing slowness was due to having a [prepare script on a git install](https://docs.npmjs.com/cli/v10/using-npm/scripts#life-cycle-scripts). `NODE_ENV=production` has no effect in this case. It is addressed with c90a465715232c16fe81d71e61509825ad07fd90 and https://github.com/nextstrain/nextstrain.org/commit/c3d38f25911dfadbbb9161d9e98f907181ab1d5e elsewhere.

**Comment by victorlin** [comment]
^ new preview PR is a bug. Will create a PR to fix.

**Pull request merged**